keep/providers/splunk_provider/splunk_provider.py

```diff
@@ -77,40 +73,57 @@ def __init__(
         super().__init__(context_manager, provider_id, config)
 
     def validate_scopes(self) -> dict[str, bool | str]:
-        list_all_objects_scope = "NOT_FOUND"
-        edit_own_object_scope = "NOT_FOUND"
+        validated_scopes = {}
+
         try:
             service = connect(
                 token=self.authentication_config.api_key,
                 host=self.authentication_config.host,
                 port=self.authentication_config.port,
             )
+
+            all_permissions = []
             for user in service.users:
                 user_roles = user.content["roles"]
                 for role_name in user_roles:
                     perms = self.__get_role_capabilities(
                         role_name=role_name, service=service
                     )
-                    if not list_all_objects_scope and "list_all_objects" in perms:
-                        list_all_objects_scope = True
-                    if not edit_own_object_scope and "edit_own_objects" in perms:
-                        edit_own_object_scope = True
-                    if list_all_objects_scope and edit_own_object_scope:
-                        break
-
-            scopes = {
-                "authenticated": True,
-                "list_all_objects": list_all_objects_scope,
-                "edit_own_objects": edit_own_object_scope,
-            }
+                    all_permissions.extend(perms)
+
+            for scope in self.PROVIDER_SCOPES:
+                if scope.name in all_permissions:
+                    validated_scopes[scope.name] = True
+                else:
+                    validated_scopes[scope.name] = "NOT_FOUND"
```

Comment:
Can you show the output of `service.users`? (as a comment or smth?) because I'm not able to understand whats `service.users` stands for - is it the users in Splunk? if so - why does the roles of the users related with the permissions of the one who connected Splunk?

Reply:
Yes, it's users in Splunk.
```
list(service.users)
PyDev console: starting.
[<splunklib.client.User object at 0xffff87e77fd0>]
```

In Splunk, token has the permissions of user who created token, this is why we checking permissions of the user